api/rest/restapi.go

```diff
@@ -666,6 +678,41 @@ func (api *API) unpinHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (api *API) pinPathHandler(w http.ResponseWriter, r *http.Request) {
+	var ci cid.Cid
+	pinPath, ps := api.parsePathOrError(w, r)
```

Comment:
with parseCidOrError, what we were checking in the if block was `ps.Cid != ""`, that wouldn't work here since cid is always "" (cause we have not resolved yet).
So we can put checks on pinPath, but there nothing that we have to check here. pinPath starts with (/ipfs | /ipns | /ipld ), that's why pinPathHandler is being executed. We can check, if the key is valid or not, but I think we decided to put those checks in cluster.go (I also find putting checks there appropriate)